src/guardian/staking-module-guard/staking-module-guard.service.ts

```diff
@@ -28,6 +28,7 @@ export class StakingModuleGuardService {
 
   private lastContractsStateByModuleId: Record<number, ContractsState | null> =
     {};
+  private duplicatedKeysWasFound = false;
```

Comment:
it's not done yet?

Reply:
Yes, about this optimization I was speaking
there are no problem to release now without it

Reply:
check for finding duplicated keys is already implemented 
there are possibility to not make it on each iteration

Reply:
check for finding duplicated keys is already implemented - and released on holesky